### PR TITLE
[0015] Occupied capacity increment rate

### PR DESCRIPTION
**Pull request by jjyr**
This PR removed the confused part from #71

### CODE REVIEW

File: rfcs/0000-occupied-capacity-increment-rate/0000-occupied-capacity-increment-rate.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+``` rust
+let mut consumed_occupied_capacity = 0;
+let mut generated_occupied_capacity = 0;
+for tx in block.commit_transactions() {
+  consumed_occupied_capacity += tx.inputs.map(|input| input.previous_output.occupied_capacity()).sum();
+  generated_occupied_capacity += tx.outputs.map(|output| output.occupied_capacity()).sum();
+}
+
+// NOTICE: this value can be negative
+assert!(generated_occupied_capacity - consumed_occupied_capacity <= MAX_INCREMENT_OCCUPIED_CAPACITY);
```
Comment (zhangsoledad) [inline rfcs/0000-occupied-capacity-increment-rate/0000-occupied-capacity-increment-rate.md]: `((generated_occupied_capacity - consumed_occupied_capacity).abs())`

File: rfcs/0000-occupied-capacity-increment-rate/0000-occupied-capacity-increment-rate.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+# Occupied capacity increment rate
+
+## Abstract
+
+Capacity is one of the core resources in CKB network, itself is just a number representing how much storage resource a user can use, once a user store something in CKB, the occupied part of storage which we called occupied capacity represents the actually used resource in the network.
+
+Occupied capacity affects CKB network because that every full node must cost the same memory or disk to save these content, it incentives miners to be more centralized which does not make any sense for a decentralized network.
+
+So the solution is to limit the increase occupied capacity in each block, to indirectly control the network's total occupied capacity and the increment rate of occupied capacity.
+
```
Comment (zhangsoledad) [inline rfcs/0000-occupied-capacity-increment-rate/0000-occupied-capacity-increment-rate.md]: You have the reason, but what's the target of the limit policy, match the current level of hardware?
Maybe we should supplement this part.

**Comment by doitian** [comment]
Is total size limit another PR?

**Comment by jjyr** [comment]
> Is total size limit another PR?

Block size limit may affect by some consensus details, so better to wait for consensus paper https://github.com/nervosnetwork/rfcs/pull/71#discussion_r256269567

File: rfcs/0000-occupied-capacity-increment-rate/0000-occupied-capacity-increment-rate.md (hunk 2 above)
Comment (jjyr) [reply rfcs/0000-occupied-capacity-increment-rate/0000-occupied-capacity-increment-rate.md]: This part should be introduced in an RFC that describes on-chain governance mechanics.

File: rfcs/0000-occupied-capacity-increment-rate/0000-occupied-capacity-increment-rate.md (hunk 1 above)
Comment (jjyr) [reply rfcs/0000-occupied-capacity-increment-rate/0000-occupied-capacity-increment-rate.md]: We should not care about the decrease of occupied_capacity.

**Comment by doitian** [comment]
Assign RFC# 0015

File: rfcs/0015-occupied-capacity-increment-rate/0015-occupied-capacity-increment-rate.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Created: 2019-2-13
+---
+
+# Occupied capacity increment rate
+
+## Abstract
+
+Capacity is one of the core resources in CKB network, itself is just a number representing how much storage resource a user can use, once a user store something in CKB, the occupied part of storage which we called occupied capacity represents the actually used resource in the network.
+
+Occupied capacity affects CKB network because that every full node must cost the same memory or disk to save these content, it incentives miners to be more centralized which does not make any sense for a decentralized network.
```
Comment (janx) [inline rfcs/0015-occupied-capacity-increment-rate/0015-occupied-capacity-increment-rate.md]: "does not make any sense" is not very descriptive.

File: rfcs/0015-occupied-capacity-increment-rate/0015-occupied-capacity-increment-rate.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+So the solution is to limit the increase occupied capacity in each block, to indirectly control the network's total occupied capacity and the increment rate of occupied capacity.
```
Comment (janx) [inline rfcs/0015-occupied-capacity-increment-rate/0015-occupied-capacity-increment-rate.md]: Per block increase capacity limit is not 'the solution' to state explosion problem. It's a fine-grained additive to CKB's token economics that bound the state size.

**Comment by janx** [comment]
I reverted the merge because the abstract is misleading. Please adjust the descriptions before merge.

File: rfcs/0015-occupied-capacity-increment-rate/0015-occupied-capacity-increment-rate.md (hunk 3 above)
Comment (janx) [reply rfcs/0015-occupied-capacity-increment-rate/0015-occupied-capacity-increment-rate.md]: It not 'incentives' but forces miners to be more centralized.

**Comment by janx** [comment]
Please also note this limit may not be necessary if the bound imposed by block cycles limit is smaller.

**Pull request merged**